cardinal/server/server_test.go

```diff
@@ -99,6 +99,84 @@ func makeTestTransactionHandler(t *testing.T, world *ecs.World, opts ...Option)
 	}
 }
 
+func TestShutDownViaMethod(t *testing.T) {
+	go func() {
+		select {
+		case <-time.After(10 * time.Second):
+			panic("took too long to shutdown.")
+		}
+	}() // If this test is frozen then it failed to shut down, create failure with panic.
+	w := inmem.NewECSWorldForTest(t)
+	sendTx := ecs.NewTransactionType[SendEnergyTx, SendEnergyTxResult]("sendTx")
+	assert.NilError(t, w.RegisterTransactions(sendTx))
+	w.AddSystem(func(world *ecs.World, queue *ecs.TransactionQueue, _ *ecs.Logger) error {
+		return nil
+	})
+	assert.NilError(t, w.LoadGameState())
+	txh := makeTestTransactionHandler(t, w, DisableSignatureVerification())
+	for !txh.IsServerRunning() {
```

Comment:
See my comment above. I believe the makeTestTransactionHandler doesn't return until the server is up and running.

Reply:
Hm I don't believe this is true. What I saw when I asserted IsServerRunning is that it actually failed. You have to wait for it. The atomics I placed are as near as possible to the running loop so it might be your function returns a little before the running loop actually occurs.

Reply:
I did some tests here. You're right I think I'm mistaken on this one.